File: eRestorante/eRestorante.Models/Model/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eRestorante.Models.Model
{
    public partial class Customer
    {
        public int CustomerId { get; set; }

        public string CustomerName { get; set; } = null!;

        public string CustomerSurname { get; set; } = null!;

        public string CustomerEmail { get; set; } = null!;

        public string CustomerPhone { get; set; } = null!;

        public DateOnly CustomerDateRegister { get; set; }

        public byte[]? CustomerImage { get; set; }

        public byte? CustomerStatus { get; set; }
    }
}

File: eRestorante/eRestoranteUI/erestorante_desktop/lib/models/customer.dart
import 'package:json_annotation/json_annotation.dart';

part 'customer.g.dart';

@JsonSerializable()
class Customer{
  int? customerId;
  String? customerName;
  String? customerSurname;
  String? customerEmail;
  String? customerPhone;
  String? customerDateRegister;
  String? customerImage;

  Customer(this.customerId, this.customerName, this.customerSurname, this.customerEmail, this.customerPhone, this.customerDateRegister, this.customerImage);
  factory Customer.fromJson(Map<String, dynamic> json) => _$CustomerFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$CustomerToJson(this);
}